zk-token-sdk/src/range_proof/errors.rs

```diff
@@ -18,6 +29,14 @@ pub enum RangeProofVerificationError {
         "attempted to verify range proof with a non-power-of-two bit size or bit size is too big"
     )]
     InvalidBitSize,
-    #[error("insufficient generators for the proof")]
-    InvalidGeneratorsLength,
+    #[error("commitments and bit lengths vectors have different lengths")]
+    VectorLengthMismatch,
+    #[error("maximum generator length exceeded")]
+    MaximumGeneratorLengthExceeded,
+}
+
+#[derive(Error, Clone, Debug, Eq, PartialEq)]
+pub enum RangeProofGeneratorError {
+    #[error("maximum generator length exceeded")]
+    MaximumGeneratorLengthExceeded,
```

Comment:
What's the purpose of this error over `RangeProofGenerationError`? They both have `MaximumGeneratorLengthExceeded`

Reply:
Yeah, both proof generation and verification uses a subroutine to generate "group generators" that could return an if the generator length is too large. I made it so that if the subroutine errors during proof generation, it returns `RangeProofGenerationError::Maximum...Exceeded` ([here](https://github.com/solana-labs/solana/pull/34065/files#diff-ec935ece57e3510c52226da542565f5dcd2bbaf887746960b12b51d5a0606081R97)) and if it errors during verification, it returns `RangeProofVerificationError::Maximum...Exceeded` ([here](https://github.com/solana-labs/solana/pull/34065/files#diff-ec935ece57e3510c52226da542565f5dcd2bbaf887746960b12b51d5a0606081R268)).